packages/use-contractkit/src/screens/wallet-connect-custom.tsx

```diff
@@ -1,72 +1,107 @@
+import QrCode from 'qrcode.react';
 import React from 'react';
+import { isMobile } from 'react-device-detect';
 import Loader from 'react-loader-spinner';
 
 import { CopyText } from '../components';
 import { useWalletConnectConnector } from '../connectors/useWalletConnectConnector';
-import { WalletIds } from '../constants';
-import { Connector } from '../types';
+import { Connector, WalletEntry } from '../types';
 
 interface Props {
   onSubmit: (connector: Connector) => void;
+  wallet: WalletEntry;
 }
 
-export const CeloWallet: React.FC<Props> = ({ onSubmit }: Props) => {
+// TODO: get in touch with CeloWallet and see if we can harmonize deeplinks
+// CeloWallet: celowallet://wc?...
+// Valora/CeloDance: celo://wallet/wc?...
+const getDeepLink =
+  (protocol = 'celo:') =>
+  (uri: string) => {
+    return `${protocol}//wc?uri=${encodeURIComponent(uri)}`;
+  };
+
+export const WalletConnectCustom: React.FC<Props> = ({
+  onSubmit,
+  wallet,
+}: Props) => {
   const uri = useWalletConnectConnector(
     onSubmit,
-    false,
-    undefined,
-    WalletIds.CeloWallet
+    isMobile,
+    getDeepLink(wallet.mobile.native),
+    wallet.id
   );
 
   const onClickPlatform = (platform: 'web' | 'desktop') => {
     if (!uri) return;
     let url;
     if (platform === 'web') {
-      url = `https://celowallet.app/wc?uri=${encodeURIComponent(uri)}`;
+      const sanitized = wallet.app.browser.replace(/\/+$/, '');
+      url = `${sanitized}/wc?uri=${encodeURIComponent(uri)}`;
     } else {
-      url = `celowallet://wc?uri=${encodeURIComponent(uri)}`;
+      url = getDeepLink(wallet.mobile.native)(uri);
```

Comment:
Does this break deep linking to Celo Wallet? Not seeing where the `celowallet` scheme gets defined now

Reply:
The deeplink is declared in WC registry ![Screenshot 2022-01-12 at 18 51 21](https://user-images.githubusercontent.com/3650621/149194887-318218d5-ec93-47c1-b211-5f7e3bbd0d54.png)